### PR TITLE
PA-2185 Updated Remove from SPL

### PR DESCRIPTION
**Pull request by Fosol**
When removing a project from SPL it now requires them to provide the following properties; `Request for removal on`, `Request for removal approved on`, and `Rationale for removal`.

image

### CODE REVIEW

File: frontend/src/constants/noteTypes.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ContractInPlace = 8,
+  /** Notes to include in reports. */
+  Reporting = 9,
+  /** Loan term information. */
+  LoanTerms = 10,
+  /** Adjustment note. */
+  Adjustment = 11,
+  /** Surplus Property Program Cost. */
+  SplCost = 12,
+  /** Surpluse Property Program Gain. */
```
Comment (devinleighsmith) [inline frontend/src/constants/noteTypes.ts:31]: Surpluse

File: frontend/src/features/projects/spl/forms/SurplusPropertyListForm.tsx [hunk 2]
```diff
@@ -198,7 +206,6 @@ const SurplusPropertyListForm = ({
           <TooltipIcon toolTipId="dateEnteredMarket" toolTip={dateEnteredMarket} />
         </Form.Label>
         <FastDatePicker
-          required
```
Comment (devinleighsmith) [inline frontend/src/features/projects/spl/forms/SurplusPropertyListForm.tsx:201]: Where these listed as required inappropriately before? This change seems to be about adding some new fields so changing the required status on some of these other date fields seems unrelated.

File: frontend/src/features/projects/spl/steps/SplStep.tsx [hunk 3]
```diff
@@ -83,8 +83,8 @@ const SplStep = ({ formikRef }: IStepProps) => {
       statusCode: DisposeWorkflowStatus.Draft,
     },
     {
-      schema: SurplusPropertyListNotInSplYupSchema,
```
Comment (devinleighsmith) [inline frontend/src/features/projects/spl/steps/SplStep.tsx:86]: it looks like the not in spl schema on the erp tab got replaced with this new schema on the spl tab, is that what is desired?

**Review by devinleighsmith — APPROVED** [review]

File: frontend/src/features/projects/spl/forms/SurplusPropertyListForm.tsx (hunk 2 above)
Comment (Fosol) [reply frontend/src/features/projects/spl/forms/SurplusPropertyListForm.tsx:201]: They're only required under certain conditions.  Doesn't really make sense to label them as always required.

File: frontend/src/features/projects/spl/steps/SplStep.tsx (hunk 3 above)
Comment (Fosol) [reply frontend/src/features/projects/spl/steps/SplStep.tsx:86]: When removing from SPL there are only 3 required fields, `Request for removal on`, `Request for removal approved on` and `Rationale for removal`.  The original validation was looking for a field on the ERP tab.

File: backend/entities/NoteTypes.cs [hunk 4]
```diff
@@ -56,11 +56,11 @@ public enum NoteTypes
         /// <summary>
         /// SPP program cost notes.
         /// </summary>
-        SppCost = 12,
+        SplCost = 12,
```
Comment (agalbrai) [inline backend/entities/NoteTypes.cs:59]: Your description still says SPP cost but the variable is SPL cost, not sure if this matters

File: frontend/src/features/projects/dispose/testUtils.ts [hunk 5]
```diff
@@ -824,140 +824,147 @@ export const mockFlatProject = {
   notes: [
     {
       id: undefined,
-      noteType: 0,
+      noteType: NoteTypes.General,
```
Comment (agalbrai) [inline frontend/src/features/projects/dispose/testUtils.ts:827]: not having to check the number 👍

File: frontend/src/features/projects/erp/steps/GreTransferStep.test.tsx [hunk 6]
```diff
@@ -22,40 +22,41 @@ const mockProject: IProject = {
...
       address: 'Test, Alert Bay',
       addressId: 1,
+      administrativeArea: 'Alert Bay',
+      province: 'British Columbia',
+      postal: '',
       agency: 'Ministry of Advanced Education, Skills & Training',
       agencyCode: 'AEST',
       agencyId: 1,
+      classification: 'Core Operational',
+      classificationId: Classifications.SurplusActive,
```
Comment (agalbrai) [inline frontend/src/features/projects/erp/steps/GreTransferStep.test.tsx:39]: Core Operational and Surplus Active?
